io.go

```diff
@@ -360,33 +364,261 @@ func (c *Cell) setEmissionsFlux(e *Emissions) {
+// }
+//
+// func checkSuffix(s string) (bool, error) {
+// 	var isSuffix bool
+// 	if string(s) != "" {
+// 		isSuffix, err := regexp.MatchString("[a-zA-Z0-9_]", string(s[len(s)-1]))
+// 		if err != nil {
+// 			return false, err
+// 		}
+// 	} else {
+// 		isSuffix = false
+// 	}
+// 	return isSuffix, nil
+// }
+
+// checkForDerivatives identifies the unique input variables that are required
+// to calculate the requested output variables.
+// Inputs:
+// (1) Map of requested output variable names to their corresponding expressions.
+// (2) Map of all function names to function definitions that are used in expressions.
+// Outputs:
+// (1) Map of output variable names to revised expressions where any user-defined
+// output variable showing up in a subsequent expression is replaced by its
+// corresponding user-defined expression.
+// (2) Slice of all unique input variables required to calculate the requested
+// output variables.
+func (o *Outputter) checkForDerivatives() error {
+	o.GetVariables = make([]string, 0, len(o.OutputVariables))
+	for key, val := range o.OutputVariables {
+		o.OutputVariables[key] = strings.Replace(val, "{", "", -1)
+		o.OutputVariables[key] = strings.Replace(o.OutputVariables[key], "}", "", -1)
+		expression, err := govaluate.NewEvaluableExpressionWithFunctions(o.OutputVariables[key], o.OutputFunctions)
+		if err != nil {
+			return fmt.Errorf("inmap OutputVariables: %v", err)
+		}
+		uniqueVars := removeDuplicates(expression.Vars())
+		o.GetVariables = append(o.GetVariables, uniqueVars...)
+		var isSuffix bool
+		var isPrefix bool
+		for _, uniqueVar := range uniqueVars {
+			if o.OutputVariables[uniqueVar] != "" && o.OutputVariables[uniqueVar] != uniqueVar {
+				splitVal := strings.Split(val, uniqueVar)
+				for i := 0; i < len(splitVal)-1; i++ {
+					if string(splitVal[i]) != "" {
+						isSuffix, err = regexp.MatchString("[a-zA-Z0-9_]", string(splitVal[i][len(splitVal[i])-1]))
+						if err != nil {
+							return fmt.Errorf("inmap OutputVariables: %v", err)
+						}
+					} else {
+						isSuffix = false
+					}
+					if string(splitVal[i+1]) != "" {
+						isPrefix, err = regexp.MatchString("[a-zA-Z0-9_]", string(splitVal[i+1][0]))
+						if err != nil {
+							return fmt.Errorf("inmap OutputVariables: %v", err)
+						}
+					} else {
+						isPrefix = false
+					}
+					splitVal[i] = splitVal[i] + uniqueVar
+					if (!isSuffix && !isPrefix) || ((len(splitVal[i]) == len(uniqueVar)) && !isPrefix) {
+						splitVal[i] = strings.Replace(splitVal[i], uniqueVar, "("+o.OutputVariables[uniqueVar]+")", -1)
+					}
+				}
+				o.OutputVariables[key] = strings.Join(splitVal, "")
+				return o.checkForDerivatives()
+			}
+		}
+	}
+	o.GetVariables = removeDuplicates(o.GetVariables)
+	return nil
+}
+
+// CheckGetNames checks whether the unique input variables required to calculate
+// the user-requested output variables are available in the model.
+func (d *InMAP) checkGetNames(g ...string) error {
```

Comment:
Is there a better name for this than "GetNames", "GetVal", etc? Maybe ModelNames or ModelVars?